functions/utils/getAsset.ts

```diff
@@ -0,0 +1,25 @@
+import { AssetDocument } from '../../src/graphql/data/__generated__/types-and-hooks'
+import { getApolloClient } from './getApolloClient'
+
+export default async function getAsset(collectionAddress: string, tokenId: string, url: string) {
+  const client = getApolloClient()
+  const { data } = await client.query({
+    query: AssetDocument,
+    variables: {
+      address: collectionAddress,
+      filter: {
+        tokenIds: [tokenId],
+      },
+    },
+  })
+  const asset = data?.nftAssets?.edges[0]?.node
+  if (!asset) {
+    return undefined
+  }
+  const formattedAsset = {
+    title: asset.name ? asset.name : asset.collection?.name + ' #' + asset.tokenId,
+    image: asset.image?.url,
```

Comment:
These images may not be available. If it is not available, do we still want to inject meta tags for images?

Reply:
I think it should be fine. Worst case we just have the default Uniswap image as the fallback image url.

Reply:
but you shouldn't render the tag at all in that case, right?